apps/passport-server/test/devconnect.spec.ts

```diff
@@ -0,0 +1,340 @@
+        },
+        {
+          email: EMAIL_1,
+          itemInfoID: 2, // Represents EVENT_B, ITEM_1
+        },
+        {
+          email: EMAIL_2,
+          itemInfoID: 2,
+        },
+        {
+          email: EMAIL_3,
+          itemInfoID: 2,
+        },
+        {
+          email: EMAIL_4,
+          itemInfoID: 2,
+        },
+        {
+          email: EMAIL_1,
+          itemInfoID: 3, // Represents EVENT_A, ITEM_2
+        },
+        {
+          email: EMAIL_2,
+          itemInfoID: 3,
+        },
+        {
+          email: EMAIL_4,
+          itemInfoID: 3,
+        },
+      ]);
+    }
+  );
+
+  step("removing an order causes soft deletion of ticket", async function () {
+    // Simulate removing order - in this instance, we remove an order from
+    // EVENT_A1 with EMAIL_1 as the purchaser that contains the only positions
+    // that have EMAIL_2 and EMAIL_3 for ITEM_1. Removing this order, then,
+    // would cause (EMAIL_2, ITEM_1) and (EMAIL_3, ITEM_1) to be soft deleted
+    // in EVENT_A.
+    const ordersForEventA = devconnectPretixMocker
+      .getMockData()
+      .ordersByEventId.get(EVENT_A_ID)!;
+    const orderWithEmail2And3 = ordersForEventA.find(
+      (o) => o.email === EMAIL_1
+    )!;
+    devconnectPretixMocker.removeOrder(EVENT_A_ID, orderWithEmail2And3.code);
+    devconnectPretixSyncService.replaceApi(
+      getDevconnectMockPretixAPI(devconnectPretixMocker.getMockData())
+    );
+
+    await devconnectPretixSyncService.trySync();
+
+    const tickets = await fetchAllDevconnectPretixTickets(
+      application.context.dbPool
+    );
+
+    // Because two tickets are removed - see comment above
+    expect(tickets).to.have.length(9);
+
+    const ticketsWithEmailEventAndItems = tickets.map((o) => ({
+      email: o.email,
+      itemInfoID: o.devconnect_pretix_items_info_id,
+    }));
+
+    expect(ticketsWithEmailEventAndItems).to.have.deep.members([
+      // Four tickets for event A because four unique emails
+      {
+        email: EMAIL_1,
+        itemInfoID: 1, // Represents EVENT_A, ITEM_1
+      },
+      // This is formerly where (EMAIL_2, ITEM_1) and (EMAIL_3, ITEM_1) were for EVENT_A
+      {
+        email: EMAIL_4,
+        itemInfoID: 1,
+      },
+      {
+        email: EMAIL_1,
+        itemInfoID: 2, // Represents EVENT_B, ITEM_1
+      },
+      {
+        email: EMAIL_2,
+        itemInfoID: 2,
+      },
+      {
+        email: EMAIL_3,
+        itemInfoID: 2,
+      },
+      {
+        email: EMAIL_4,
+        itemInfoID: 2,
+      },
+      {
+        email: EMAIL_1,
+        itemInfoID: 3, // Represents EVENT_A, ITEM_2
+      },
+      {
+        email: EMAIL_2,
+        itemInfoID: 3,
+      },
+      {
+        email: EMAIL_4,
+        itemInfoID: 3,
+      },
+    ]);
+  });
+
+  // TODO: More tests
+  // 1. Test that item_name in ItemInfo and event_name EventInfo always syncs with Pretix.
+  // 2. Test deleting positions within orders (not just entire orders).
```

Comment:
another feature to make sure we support is cancelling events/positions (both are possible) should soft delete the ticket in our database as well. 
another category of tests worth noting is testing that the sync code checks that events are configured in pretix according to our instructions - eg don't send a ticket in an email, and that the attendee_email is collected per-product
we'll think of more tests later too.